src/Features/Core/Portable/DocumentHighlighting/AbstractDocumentHighlightsService.cs

```diff
@@ -73,6 +76,14 @@ internal abstract partial class AbstractDocumentHighlightsService : IDocumentHig
             // Get unique tags for referenced symbols
             return await GetTagsForReferencedSymbolAsync(
                 symbol, document, documentsToSearch, cancellationToken).ConfigureAwait(false);
+
+            bool IsBindableNonOperator(SyntaxToken token)
+            {
+                // If the cursor is at the end of an identifier, with an operator immediately following, eg count$$++;
+                // the normal logic would be to return the operators semantic info, but that isn't useful for highlighting
+                // so we exclude operators from being valid matches.
+                return syntaxFacts.IsBindableToken(token) && !syntaxFacts.IsOperator(token);
```

Comment:
i think we *only* want this if there's a selection.  if there's not a selection, this is the intended and intentional behavior as we want people to be able to go to operators.  in other words, if i have `a+b`, then `$$a+b` goes to `a`, `a$$+b` goes to `+`.

However, if we have `[|a|]+`, then we should go to `a` not `+`.  Another way to support this would be to have the caller pass in the *start* of hte span, and not the end of it.

Reply:
> we want people to be able to go to operators

Do you mean when they use Go To Def? Because that functionality is unchanged by this, that's why the change is in the document highlight service specifically, and not the symbol finder. With this change given `a$$+b` then all `a`s in the scope will be highlighted, but pressing F12 will go to the `+` operator.

Additionally, a selection is not required to repro this issue, nor an existing highlight. Simply type `var count; count++;` and put your cursor directly in front of the + and you'll see nothing is highlighted at all.

Reply:
Unless you mean it should highlight the other instances of that operator, so that go to next/prev highlight work? That certainly doesn't happen visually at the moment, but would be an alternate solution.